src/cdk/datetime/datetime.md

```diff
@@ -0,0 +1,111 @@
+A `CdkDatepicker` deals with providing a uniform datepicker amongst different datepicker implementations. Whenever a
+directive needs to read datepicker information, the `CdkDatepicker` will provide the properties needed. Users will be
+able to flexibly utilize the `CdkDatepicker` to custom create their own datepicker or calendar.
+
+A `CdkDatepickerInput` is a directive that's used to decorate a native input so that it can work with a `CdkDatepicker`.
+The `CdkDatepickerInput` provides optional input to the datepicker to provide for different datepicker implementations:
+ * Minimum & maximum dates - The minimum and maximum dates can be set by the `CdkDatepickerInput` and will update the
+ calendar accordingly to clamp the dates of the datepicker.
+ * Date filter - Given a function set by the `CdkDatepickerInput`, the date filter will evaluate that function and
+ update the calendar accordingly to filter out dates of the datepicker.
+ * Disabled - This input provides the option to disable the `CdkDatepickerInput` directive from feeding inputs into the
+ datepicker.
+
+Each `CdkDatepicker` must be associated with a `CdkDatepickerInput` to work properly.
+
+A reference material datepicker implementation is available from the npm package `@angular/material/datepicker`.
+
+#### Building a custom calendar that works with the `CdkDatepicker`
+ * Create a component that extends `CalendarView` and implement all abstract properties:
+    * The minimum, maximum, selected, and active dates are initially set by the datepicker and will update the calendar
+    accordingly.
+    * The selected date can be set by the calendar. The selected date will emit on a stream when the date has changed,
+    and the datepicker will subscribe to this stream and update accordingly.
+ * Emit on `selectedChange` from the `CalendarView` when the date has changed.
+ * Provide the newly created component as the `CalendarView` injection token in the component decorator.
+
+```ts
+@Component({
+  providers: [{provide: CalendarView, useExisting: MyCalendar}],
+  ...
+})
+```
+
+ <!-- example(cdk-datepicker-overview) -->
+
+#### Custom calendar on top of `CdkDatepicker` example
+The user can create their own calendar extended from `CalendarView` with all of its abstract properties satisfied:
+
+```ts
+@Component({
+  providers: [{provide: CalendarView, useExisting: MyCalendar}],
+  ...
+})
+export class MyCalendar<D> extends CalendarView<D> {
+  activeDate = ...
+  minDate = ...
+  maxDate = ...
+  selected = ...
+  
+  constructor(dateAdapter: DateAdapter<D>) {
+    super();
+}
+```
+
+Whenever the selected date has changed, the user can also use the `selectedChange` in `CalendarView` to emit when the
+date has changed:
+
+```ts
+selectedChange.emit(newDate);
+```
+
+Finally, the user can use their custom `MyCalendar` component:
+
+```html
+<input [cdkDatepicker]="picker">
+<cdk-datepicker #picker><my-calendar (selectedChange)="..."></my-calendar></cdk-datepicker>
+```
```

Comment:
I would copy this code snippet up right after the text under `### `CdkDatepicker` Component`. The whole concept is much more clear with the template code.

Reply:
done